src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -125,7 +125,7 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       "JOIN prison p ON p.id = v.prison_id " +
       "JOIN session_template st ON st.reference = v.session_template_reference " +
       "WHERE p.code = :prisonCode AND " +
-      "Date(v.visit_start) = :sessionDate AND " +
+      "(v.visit_start >= :sessionDate AND v.visit_start < CAST(:sessionDate AS DATE) + CAST('1 day' AS INTERVAL)) AND " +
       "st.start_time = :sessionStartTime AND " +
       "st.end_time = :sessionEndTime AND " +
```

Comment:
Also the sessionStartTime and endtime check might not be needed anymore.

Reply:
```
"st.start_time = :sessionStartTime AND " +
"st.end_time = :sessionEndTime AND " +
```
Is key to the logic I dont understand why we would remove it?

Reply:
Your right, all I need is the session template ref